cmd/process-agent/main.go

```diff
@@ -4,11 +4,49 @@ package main
 
 import (
 	"flag"
+	"github.com/DataDog/datadog-agent/cmd/agent/common"
+	"github.com/DataDog/datadog-agent/pkg/api/util"
+	"log"
 	_ "net/http/pprof"
+	"strings"
 
 	"github.com/DataDog/datadog-agent/cmd/process-agent/flags"
 )
 
+func setupConfig() {
+	verb, key, value :=
+		strings.ToLower(flag.Arg(1)),
+		strings.ToLower(flag.Arg(2)),
+		flag.Arg(3)
+
+	// Configure a session token so nothing gets rejected
+	err := util.SetAuthToken()
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+	settingsClient, err := common.NewSettingsClient()
+
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+
+	if verb == "get" {
+		get, err := settingsClient.Get(key)
+		if err == nil {
+			log.Println(get)
+		}
+	} else if verb == "set" {
+		_, err = settingsClient.Set(key, value)
+	} else {
+		log.Println("Usage: process-agent config {get setting, set setting value}")
+	}
```

Comment:
Maybe I'm missing some context, but how is this allowing us to toggle the log level?

Reply:
@just-chillin Let me know if my understanding is correct:

In `main()`, we check whether the command passed to the process-agent is `config`, and if so, it sets `opt.IsConfig = true`. Then in `runAgent()`, if `opt.IsConfig == true`, we run this function to interface with the settings client, set or get a setting based on the command-line args, and exit.

Reply:
Yep! That's it exactly